File: sentimental-cash/cash.py
from cs50 import get_float

while True:
    cents = get_float("Change owed: ")
    # Check if input more than 0
    if cents > 0:
        break


# Count numbers of quaters
def calculate_quaters(cents):
    if cents >= 0.25:
        quaters = cents / 0.25
    else:
        quaters = 0
    return int(quaters)


# Count number of dimes
def calculate_dimes(cents):
    if cents >= 0.10:
        dimes = cents / 0.10
    else:
        dimes = 0
    return int(dimes)


# Count number of nickels
def calculate_nickels(cents):
    if cents >= 0.05:
        nickels = cents / 0.05
    else:
        nickels = 0
    return int(nickels)


# Count number of pennies
def calculate_pennies(cents):
    if cents >= 0.01:
        pennies = cents / 0.01
    else:
        pennies = 0
    return int(pennies)


# Call function and assing return to quaters variable
quaters = calculate_quaters(cents)
cents = round((cents - quaters * 0.25), 2)
# Call function and assing return to dimes variable
dimes = calculate_dimes(cents)
cents = round((cents - dimes * 0.10), 2)
# Call function and assing return to nickels variable
nickels = calculate_nickels(cents)
cents = round((cents - nickels * 0.05), 2)
# Call function and assing return to pennies variable
pennies = calculate_pennies(cents)
# Print result
print(quaters + dimes + nickels + pennies)

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents();
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed

    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}
//Request users to input number of cents
int get_cents()
{
    int cents;
    do
    {
        cents = get_int("How many cents we own you?:");
    }
    while (cents < 0);
    return cents;
}
//Calculate number of quarters
int calculate_quarters(int cents)
{
    int quarters;
    if (cents >= 25)
    {
        quarters = cents / 25;
    }
    else
    {
        quarters = 0;
    }
    return quarters;
}
//Calculate number of dimes
int calculate_dimes(int cents)
{
    int dimes;
    if (cents >= 10)
    {
        dimes = cents / 10;
    }
    else
    {
        dimes = 0;
    }
    return dimes;
}
//Calculate number of nickels
int calculate_nickels(int cents)
{
    int nickels;
    if (cents >= 5)
    {
        nickels = cents / 5;
    }
    else
    {
        nickels = 0;
    }
    return nickels;
}
//Calculate number of pennies
int calculate_pennies(int cents)
{
    int pennies;
    if (cents >= 1)
    {
        pennies = cents / 1;
    }
    else
    {
        pennies = 0;
    }
    return pennies;
}
